### PR TITLE
[1.6.x] Handling single valid item AllOf schema in client/service generation

### PR DESCRIPTION
**Pull request by aneeshafedo**
Resolves https://github.com/ballerina-platform/openapi-tools/issues/1511 in 1.6.x 

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: openapi-cli/src/test/resources/generators/diagnostic_files/single_allOf.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      description: Optional extended description in Markdown.
+      parameters:
+        - name: id
+          in: path
+          description: ID of user to fetch
+          required: true
+          schema:
+            allOf:
+              - $ref: "#/components/schemas/Name"
+              - description: "Name details"
```
Comment (NipunaRanasinghe) [inline openapi-cli/src/test/resources/generators/diagnostic_files/single_allOf.yaml:22]: Will we need a negative test case to evaluate the output, when an allOf schema is having no valid types? :)

Comment (aneeshafedo) [reply openapi-cli/src/test/resources/generators/diagnostic_files/single_allOf.yaml:22]: Addressed in https://github.com/ballerina-platform/openapi-tools/pull/1521/commits/e77dd15e8ff5a233bf35281b2e335fc87cf01a34 to generate the type as `anydata` in such scenarios

File: openapi-core/src/main/java/io/ballerina/openapi/core/generators/schema/ballerinatypegenerators/AllOfRecordTypeGenerator.java [hunk 2]
```diff
@@ -103,23 +103,37 @@ public TypeDescriptorNode generateTypeDescriptorNode() throws BallerinaOpenApiEx
...
-                            createToken(CLOSE_BRACE_PIPE_TOKEN));
+            ImmutablePair<List<Node>, List<Schema<?>>> recordFlist = generateAllOfRecordFields(allOfSchemas);
+            List<Node> recordFieldList = recordFlist.getLeft();
+            List<Schema<?>> validSchemas = recordFlist.getRight();
+            if (validSchemas.size() == 1) {
+                TypeGenerator typeGenerator = getTypeGenerator(validSchemas.get(0), typeName, null);
+                return typeGenerator.generateTypeDescriptorNode();
+            } else if (validSchemas.isEmpty()) {
+                AnyDataTypeGenerator anyDataTypeGenerator = new AnyDataTypeGenerator(schema, typeName);
+                return anyDataTypeGenerator.generateTypeDescriptorNode();
```
Comment (NipunaRanasinghe) [inline openapi-core/src/main/java/io/ballerina/openapi/core/generators/schema/ballerinatypegenerators/AllOfRecordTypeGenerator.java:114]: Shall we change the order of these cases to be incremental?

Comment (aneeshafedo) [reply openapi-core/src/main/java/io/ballerina/openapi/core/generators/schema/ballerinatypegenerators/AllOfRecordTypeGenerator.java]: Updated in fc2dde1

**Pull request merged**